receiver/otelarrowreceiver/config.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelarrowreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+const (
+	// Protocol values.
+	protoGRPC = "protocols::grpc"
+	protoHTTP = "protocols::http"
+)
+
+type httpServerSettings struct {
+	*confighttp.HTTPServerSettings `mapstructure:",squash"`
+
+	// The URL path to receive traces on. If omitted "/v1/traces" will be used.
+	TracesURLPath string `mapstructure:"traces_url_path,omitempty"`
+
+	// The URL path to receive metrics on. If omitted "/v1/metrics" will be used.
+	MetricsURLPath string `mapstructure:"metrics_url_path,omitempty"`
+
+	// The URL path to receive logs on. If omitted "/v1/logs" will be used.
+	LogsURLPath string `mapstructure:"logs_url_path,omitempty"`
+}
+
+// Protocols is the configuration for the supported protocols.
+type Protocols struct {
+	GRPC  *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
+	HTTP  *httpServerSettings            `mapstructure:"http"`
+	Arrow *ArrowSettings                 `mapstructure:"arrow"`
```

Comment:
Does this need a separate section for configuration? If the only setting is a memory limit, any reason not to include it at the top level?

Reply:
Relates to https://github.com/open-telemetry/otel-arrow/issues/43. This code is designed to drop-in where an OTLP receiver once stood, so leaving the Arrow settings in a separate section for future compatibility (was the idea).